src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/TransformationRequestCreator.java

```diff
@@ -1,12 +1,39 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.client;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.DocumentType;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.DocumentUrl;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.OcrDataField;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.ScannedDocument;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.request.TransformationRequest;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.model.internal.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.envelopes.model.Document;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.envelopes.model.Envelope;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
 
 @Service
 public class TransformationRequestCreator {
 
+    private final String documentManagementUrl;
+    private final String documentManagementContextPath;
+
```

Comment:
🤔 why document url and path are part of TransformationRequestCreator ?

Reply:
Because it needs to build document URLs and for that it needs these two pieces of data.

Reply:
Is it not processor has already uploaded documents to document management so whole url should be available to orchestrator ? why does orchestrator needs to create it again ?

Reply:
I think initially the processor used to pass the full URL to the orchestrator, but we were requested to change it and now the processor passes document IDs to the orchestrator instead.